python/pyiceberg/schema.py

```diff
@@ -276,6 +279,32 @@ def primitive(self, primitive: PrimitiveType) -> T:
         """Visit a PrimitiveType"""
 
 
+class PreOrderSchemaVisitor(Generic[T], ABC):
```

Comment:
Is this pre-order? In Java we called it `CustomOrder` because you can choose when to visit children by accessing the callable.

Reply:
It is pre-order traversal since we start at the root and then move to the leaves. In order is a bit less intuitive since it is not a binary tree. You could also do a reverse in-order, but not sure if we need that. We can also call it CustomOrder if you have a strong preference, but I think pre-order is the most logical way of using this visitor.